tests/robustness/multiclient/client/client.go

```diff
@@ -0,0 +1,70 @@
+// +build darwin,amd64 linux,amd64
+
+// Package client manages client specific info
+package client
+
+import (
+	"context"
+	"sync"
+
+	petname "github.com/dustinkirkland/golang-petname"
+	"github.com/google/uuid"
+)
+
+type key int
+
+const (
+	clientKey key = 0
+	nameLen   int = 2
+)
+
+// Client is a unique client for use in multiclient robustness tests.
+type Client struct {
+	ID string
+}
+
+func newClient() *Client {
+	petname.NonDeterministicMode()
+
+	return &Client{
+		ID: petname.Generate(nameLen, "-") + "-" + uuid.NewString(),
+	}
+}
+
+// WrapContext returns a copy of ctx with a new client.
+func WrapContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, clientKey, newClient())
+}
+
+// WrapContexts returns copies of ctx with n new clients.
+func WrapContexts(ctx context.Context, n int) []context.Context {
+	ctxs := make([]context.Context, n)
+	for i := range ctxs {
+		ctxs[i] = WrapContext(ctx)
+	}
+
+	return ctxs
+}
+
+// UnwrapContext returns a client from the given context.
+func UnwrapContext(ctx context.Context) (*Client, bool) {
+	c, ok := ctx.Value(clientKey).(*Client)
+	return c, ok
+}
```

Comment:
There is no added value in returning the bool.  I suggest
```
func UnwrapContext(ctx context.Context) *Client {
	c, _ := ctx.Value(clientKey).(*Client)
	return c
}
```

Reply:
I just tested this and I do think I need the bool there. The reason is that if anyone tries to run this function on a context that has not been wrapped with a client, they will get `nil` and then the type assertion panics:

```
panic: interface conversion: interface {} is nil, not *client.Client
```

It seems friendlier to just have an `ok` value that users can check after calling `UnwrapContext`.

Reply:
Exactly! My point was that one has to test any return value anyway so they might as well test for not nil.

Reply:
I'll update the PR to incorporate this change. I'm just realizing now that only the single-return form of the type assertion will panic. So using the two-return form with a `_` variable as you suggested above won't panic and I can just check for a `nil` return when using `UnwrapContext`.